openmdao/core/problem.py

```diff
@@ -2103,39 +2103,121 @@ def load_case(self, case):
 
         Parameters
         ----------
-        case : Case object
-            A Case from a CaseRecorder file.
+        case : Case or dict
+            A Case from a CaseReader, or a dictionary with key 'inputs' mapped to the
+            output of problem.model.list_inputs and key 'outputs' mapped to the output
+            of prob.model.list_outputs. Both list_inputs and list_outputs should be called
+            with `prom_name=True` and `return_format='dict'`.
         """
-        inputs = case.inputs if case.inputs is not None else None
-        abs2idx = self.model._var_allprocs_abs2idx
+        model = self.model
+
+        # if model overrides load_case, then call the overloaded method
+        if overrides_method('load_case', model, System):
+            model.load_case(case)
+            return
+
+        # find all subsystems that override the load_case method
+        system_overrides = {}
+        for subsys in model.system_iter(include_self=False, recurse=True):
+            if overrides_method('load_case', subsys, System):
+                system_overrides[subsys.pathname] = subsys
+
+        def set_later(var_name):
+            # determine if variable should be set later via an overridden load_case method
+            for pathname in system_overrides:
+                if var_name.startswith(pathname):
```

Comment:
Need to put a '.' after pathname to be certain that a variable lives under that system.